src/Router/types.ts

```diff
@@ -0,0 +1,49 @@
+import { ComponentType } from "react"
+import { RouteConfig } from "found"
+import { ContextProps } from "../Components/Artsy"
+
+type ReactComponent = ComponentType<any>
+type HistoryProtocol = "browser" | "hash" | "memory"
+
+export interface AppConfig {
+  historyProtocol?: HistoryProtocol
+  routes: RouteConfig
+  url?: string
+  user?: User
+}
+
+export interface ClientResolveProps {
+  ClientApp: ReactComponent
+}
+
+export interface ServerResolveProps {
+  ServerApp?: ReactComponent
+  redirect?: string
+  status?: string
+}
+
+export interface Router {
+  routes: RouteConfig
+  resolver: any // FIXME
+}
+
+export interface AppShellProps {
+  loadableState?: {
+    getScriptTag: () => string
+  }
+  data?: Object[]
```

Comment:
This should probably be `object[]`, very rarely should you use the uppercase version of these types.